### PR TITLE
Add new flag and fix some issues

### PR DESCRIPTION
**Pull request by xtomyserrax**
Let me know if it needs changes or you want anything to work differently.

The flag I added is: PREVENT_SPAWN_EGG (will prevent non allowed players in a field to spawn mobs with eggs, to bypass this Im using the same bypass permission for use as they are related)

### CODE REVIEW

**Comment by xtomyserrax** [comment]
I just tested it and it works fine! Ill wait though for your okay before merging, maybe you have any suggestion

**Comment by xtomyserrax** [comment]
I also added a bug fix, players could place a PREVENT_ENTRY field and if you were inside and not allowed you would get stuck. Now they will get teleported out

**Comment by xtomyserrax** [comment]
The latest change I made let me know if you agree, but there is actually a setting that lets you disable players from breaking fields blocks in creative mode. But actually it just prevents the field from be removed, but the block itself is destroyed and then it gets really confusing.

So I just cancelled the event if this feature was enabled and also sent a feedback message so players know what is going on there.

File: src/main/java/net/sacredlabyrinth/Phaed/PreciousStones/listeners/PSPlayerListener.java [hunk 1]
```diff
@@ -541,6 +541,24 @@ public void onPlayerInteract(PlayerInteractEvent event) {
                 }
             }
         }
+        
+        // -------------------------------------------------------------------------------- trying to spawn mobs/animals with spawn egg
+
+		if (is.getType().name().contains("_EGG")) {
```
Comment (NathanWolf) [inline src/main/java/net/sacredlabyrinth/Phaed/PreciousStones/listeners/PSPlayerListener.java]: It feels like there ought to be a nicer way to do this- I wonder if we could do like 
`if (is.getItemMeta() instanceof SpawnEggMeta)` 
?

**Review by NathanWolf — CHANGES_REQUESTED** [review]
I had one comment about hoping there's a cleaner way to check if an item is a spawn egg, but otherwise I see nothing wrong.

I'm going to want to squash this all when I merge it- so in the future it'd be better to keep changes separate if possible, like the "fix some issues" probably should've been a separate PR from the new flag, but it's not a huge deal for now.

Thank you!

File: src/main/java/net/sacredlabyrinth/Phaed/PreciousStones/listeners/PSPlayerListener.java (hunk 1 above)
Comment (xtomyserrax) [reply src/main/java/net/sacredlabyrinth/Phaed/PreciousStones/listeners/PSPlayerListener.java]: Hey, yeah, that is a really nicer and smarter way! Ty! Done

**Comment by xtomyserrax** [comment]
Yeah will make differents PRs then. I just made all in one because its easier for me to edit everything in one branch but I totally understand your point! Thanks for checking the code :D

**Review by NathanWolf — APPROVED** [review]
Oh great, so that worked? Were you able to test it? Thanks!

**Comment by xtomyserrax** [comment]
Yeah I just tested and it works perfectly fine!

**Comment by NathanWolf** [comment]
Thanks! 
Sadly the build failed- I'll take a look when I get a chance
```
ERROR] /var/lib/jenkins/workspace/PreciousStones/src/main/java/net/sacredlabyrinth/Phaed/PreciousStones/listeners/PSPlayerListener.java:[547,34] error: cannot find symbol
```

http://jenkins.elmakers.com/job/PreciousStones/29/net.sacredlabyrinth.Phaed$PreciousStones/console

**Comment by xtomyserrax** [comment]
Sorry, I forgot to add the import of the new SpawnEggMeta (it was on my eclipse when I tested but I just copied and pasted that line)

**Comment by xtomyserrax** [comment]
Issue fixed: http://jenkins.elmakers.com/job/PreciousStones/30/

**Comment by NathanWolf** [comment]
Ah, ok! I can just add it rather than going through another PR. Thanks!

**Comment by NathanWolf** [comment]
Oops thanks I was too slow XD

**Comment by xtomyserrax** [comment]
Yeah I always forget that github is just text xD. So I basicly add the code thinking as eclipse would do that it auto imports lol

**Pull request merged**